composer.json

```diff
@@ -8,7 +8,8 @@
         "guzzlehttp/psr7": "^1.0",
         "react/socket": "^0.4",
         "react/stream": "^0.4",
-        "evenement/evenement": "^2.0"
+        "evenement/evenement": "^2.0",
+        "react/promise": "^2.0 || ^1.0"
```

Comment:
Because the sink relies on promises and since 2.1 and 1.2 both introduce cancelable promises plus their API is compatible either is required for wider compatibility. When @jsor releases 3.0 I'll have to research it to see how compatible it is to use, and when it is 3.0, 2.1, and 1.2 will be targeted. This lets the decision which version to use up to composer and the project that is requiring this package which to use.